rest/blip_client_test.go

```diff
@@ -573,11 +582,25 @@ func (btr *BlipTesterReplicator) sendMsg(msg *blip.Message) (err error) {
 }
 
 // PushRev creates a revision on the client, and immediately sends a changes request for it.
-// The rev ID is always: "N-abcxyz", where N is rev generation for predictability.
+// The rev ID is always: "N-abc", where N is rev generation for predictability.
 func (btc *BlipTesterClient) PushRev(docID, parentRev string, body []byte) (revID string, err error) {
+	return btc.PushRevWithHistory(docID, parentRev, body, 1, 0, false)
+}
 
-	// generate fake rev for gen+1
+// PushRevWithHistory creates a revision on the client with history, and immediately sends a changes request for it.
+func (btc *BlipTesterClient) PushRevWithHistory(docID, parentRev string, body []byte, revCount, revpos int, skipDocNotFoundErr bool) (revID string, err error) {
```

Comment:
As discussed, change skipDocNotFoundErr to 'prunedRevisionCount', and have the test pass parentRev="" and prunedRevisionCount = 1 for the scenario where we want to send a pruned history.  Line 593 would calculate revGen as parentRevGen+revCount+prunedRevisionCount.

Reply:
Sure, the PushRevWithHistory has been refactored and included an additional method to store the revision only on the client-side.